lib/withOnyx.d.ts

```diff
@@ -0,0 +1,23 @@
+import {IsEqual} from 'type-fest';
+import {Key, CollectionKey, Selector, Value} from './types';
+
+type Mapping<TComponentProps, TOnyxProps, TMappingKey extends keyof TOnyxProps, TOnyxKey extends Key | CollectionKey> = IsEqual<Value[TOnyxKey] | null, TOnyxProps[TMappingKey]> extends true
+    ? {
+          key: TOnyxKey | ((props: Omit<TComponentProps, keyof TOnyxProps>) => TOnyxKey);
+          canEvict?: boolean | ((props: Omit<TComponentProps, keyof TOnyxProps>) => boolean);
+          initWithStoredValues?: boolean;
+          selector?: Selector<TOnyxKey>;
```

Comment:
Selector influences the prop that component gets, right? So if we use a selector that transform onyx value like here:
```ts
    report2: {
        key: `${ONYXKEYS.COLLECTION.REPORT}${'report2'}`,
        selector: 'data.isRead',
    },
```

This way report2 in the component should be `boolean` and not `Report`.

```ts
function Component({prop1, prop2 = 2, report2}: Props) {
                                                                      // ^ report2: Report | null // should be boolean
```

cc @fabioh8010

Reply:
Actually @hayata-suenaga mentioned it [here](https://github.com/Expensify/react-native-onyx/pull/267#discussion_r1263092472).

Reply:
Yes, I'm checking that.